packages/components/header/template.njk

```diff
@@ -1,96 +1,82 @@
+    {%- else -%}
+      <a class="nhsuk-header__link{% if params.service %} nhsuk-header__link--service {% endif %}" href="{{ params.homeHref | default('/') }}" aria-label="{{ params.ariaLabel | default('NHS homepage') }}">
         {{ nhsLogo | safe }}
         {%- if params.service %}
-        <span class="nhsuk-header__service-name">
-          {{ params.service.name }}
-        </span>
+        <span class="nhsuk-header__service-name">{{ params.service.name }}</span>
         {%- endif %}
       </a>
-      {%- endif %}
+    {%- endif %}
     </div>
 
-{%- if showNav == "false" and showSearch == "false"%}
-  {%- if params.transactionalService%}
+{%- if params.transactionalService %}
     <div class="nhsuk-header__transactional-service-name">
-      <a class="nhsuk-header__transactional-service-name--link" href="{% if params.transactionalService.href %}{{ params.transactionalService.href }}{% else %}/{% endif %}">{{ params.transactionalService.name }}</a>
+      <a class="nhsuk-header__transactional-service-name--link" href="{{ params.transactionalService.href | default('/') }}">{{ params.transactionalService.name }}</a>
     </div>
-  {%- endif %}
-{% endif -%}
+{%- endif %}
 
-{% if showSearch == "true" %}
+{%- if params.search %}
     <div class="nhsuk-header__content" id="content-header">
       <div class="nhsuk-header__search">
-          <div class="nhsuk-header__search-wrap" id="wrap-search">
-            <form class="nhsuk-header__search-form" id="search" action="{{ params.searchAction if params.searchAction else 'https://www.nhs.uk/search/' }}" method="get" role="search">
-              <label class="nhsuk-u-visually-hidden" for="search-field">Search the NHS website</label>
-              <input class="nhsuk-search__input" id="search-field" name="{{ params.searchInputName if params.searchInputName else "search-field" }}" type="search" placeholder="Search" autocomplete="off" >
-              <button class="nhsuk-search__submit" type="submit">
-                <svg class="nhsuk-icon nhsuk-icon__search" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" aria-hidden="true" focusable="false">
-                  <path d="M19.71 18.29l-4.11-4.1a7 7 0 1 0-1.41 1.41l4.1 4.11a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42zM5 10a5 5 0 1 1 5 5 5 5 0 0 1-5-5z"></path>
-                </svg>
-                <span class="nhsuk-u-visually-hidden">Search</span>
-              </button>
-            </form>
-          </div>
+        <div class="nhsuk-header__search-wrap" id="wrap-search">
+          <form class="nhsuk-header__search-form" id="search" action="{{ params.search.action | default('https://www.nhs.uk/search/') }}" method="get" role="search">
+            <label class="nhsuk-u-visually-hidden" for="search-field">{{ params.search.visuallyHiddenLabel | default('Search the NHS website') }}</label>
```

Comment:
Wonder if it’s worth moving these defaults to the top of the template file and using a bunch of `set variableName = ...` statements, just to make it a bit clearer what all the text strings are that can be set? Just an idea though.

Reply:
Done. Also added 2 extra options: `search.placeholder` (defaults to ‘Search’) and `search.visuallyHiddenButton` (defaults to ‘Search’). That pretty much covers everything you might want to customise for the search input.